pyfar/dsp/fft.py

```diff
@@ -292,25 +395,138 @@ def _is_odd(num):
     return bool(num & 0x1)
 
 
-def _n_bins(n_samples):
+def _n_bins_from_n_samples(n_samples, complex_time=False):
     """
     Helper function to calculate the number of bins resulting from a FFT
     with n_samples
 
-    Paramters
-    ---------
+    Parameters
+    ----------
     n_samples : int
         Number of samples
+    complex : bool
+        Flag which indicates if the time data are real or complex-valued.
 
     Returns
     -------
     n_bins : int
         Resulting number of frequency bins
 
     """
-    if _is_odd(n_samples):
-        n_bins = (n_samples+1)/2
+
+    return int(n_samples) if complex_time else n_samples // 2 + 1
+
+
+def _n_samples_from_n_bins(num_freq_bins, is_complex=False):
+    """
+    Helper function to calculate the number of samples resulting from
+    an inverse FFT of a spectrum with n_freq_bins
+
+    Parameters
+    ----------
+    num_freq_bins : int
+        Number of frequency bins
+    complex : bool
+        Flag which indicates if the frequency data are a one or two-sided
+        spectrum.
+
+    Returns
+    -------
+    n_samples : int
+        Resulting number of time samples.
+
+    """
+    return num_freq_bins if is_complex else max(1, (num_freq_bins - 1) * 2)
+
+
+def _check_conjugate_symmetry(data):
+    """
+    Check if the frequency bins are conjugate symmetric
+    around 0 Hz.
+
+    Parameters
+    -------
+    data : numpy array
+        M-dimensional array of double-sided spectrum of shape (..., N)
+        containing N frequency bins. The 0 Hz bin must always be at index
+        `data.shape[-1] // 2`.
+
+    Returns
+    -------
+    results : bool
+        return `True` if the fequency data are conjugate symmetric around
+        0 Hz, return `False` if not.
+
+    """
+    dc_idx = data.shape[-1] // 2
+    if _is_odd(data.shape[-1]):
+        mirror_spec = np.conj(np.flip(data[..., :dc_idx], axis=-1))
+    else:
+        mirror_spec = np.conj(np.flip(data[..., 1:dc_idx], axis=-1))
+
+    return bool(mirror_spec.shape[-1] > 0 and np.allclose(
+            data[..., dc_idx+1:], mirror_spec, rtol=1e-15))
```

Comment:
- Same comment as above on the tolerance. 
- Why is the functionality implemented here and in the object? Can't the function be re-used?